src/json.zig

```diff
@@ -2,6 +2,28 @@ const std = @import("std");
 
 const protobuf = @import("protobuf.zig");
 
+/// Options for protobuf JSON serialization/deserialization
+pub const Options = struct {
+    /// When true (default), oneof fields are serialized with a wrapper object
+    /// containing the oneof field name. For example:
+    ///
+    /// message AnyValue { oneof value { string string_value = 1; } }
+    ///
+    /// With emit_oneof_field_name = true (default):
+    ///   {"value":{"stringValue":"hello"}}
+    ///
+    /// With emit_oneof_field_name = false (standard protobuf JSON mapping):
+    ///   {"stringValue":"hello"}
+    ///
+    /// Set to false for standard protobuf JSON mapping where oneof union fields
+    /// are flattened into the parent message object.
+    emit_oneof_field_name: bool = true,
+};
+
+/// Global options for protobuf JSON serialization/deserialization.
+/// These can be modified at runtime before calling parse/stringify.
+pub var pb_options: Options = .{};
```

Comment:
global objects may cascade in very tricky to solve issues. could you refactor this to be passed on as options to the json (de)serializer?